src/main/java/org/zalando/failsafeactuator/config/FailsafeAutoConfiguration.java

```diff
@@ -30,9 +33,18 @@
 
   private CircuitBreakerRegistry circuitBreakerRegistry;
 
+  private MetricRegistry metricRegistry;
+
+  @Bean
+  public MetricRegistry metricRegistry() {
+    this.metricRegistry = new MetricRegistry();
+    return this.metricRegistry;
+  }
+
   @Bean
+  @DependsOn("metricRegistry")
   public CircuitBreakerRegistry circuitBreakerRegistry() {
-    circuitBreakerRegistry = new CircuitBreakerRegistry();
+    circuitBreakerRegistry = new CircuitBreakerRegistry(this.metricRegistry);
```

Comment:
You shouldn't assign to instance variables here, but rather take `metricsRegistry` as a parameter.